### ISSUE TITLE
minitensor: Do not use deprecated `KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_*` macros

### ISSUE THREAD

**Issue by dalg24** [trilinos/Trilinos#11511]
## Bug Report
@lxmota
@ndellingwood 

Compatibility with Kokkos

### Description
See kokkos/kokkos#4668 or (better) the [release 3.6 briefing slides](https://github.com/kokkos/kokkos-tutorials/blob/main/Other/ReleaseBriefings/release-36.pdf) p45-50

Please make sure you read the warning on the last page.  Don't just go and replace with `KOKKOS_IF_ON_{HOST,DEVICE}`, consider whether you can deduce the execution space and specialize or use `if constexpr`.

These are not portable and we are about to remove them.

```
packages/minitensor/src/MiniTensor_TensorBase.i.h:402:#ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST

**Comment by lxmota** [comment]
I am not familiar with Kokkos. When I wrote this library it did not contain any Kokkos code, which was later added by others. I am not sure how to fix this.

**Comment by dalg24** [comment]
OK I suggested a fix in #11613.  Please review it.
